src/python/pants/core/subsystems/python_bootstrap.py

```diff
@@ -69,6 +69,7 @@ class PythonBootstrapSubsystem(Subsystem):
         ),
         advanced=True,
         metavar="<binary-paths>",
+        environment_sensitive=True,
```

Comment:
`environment` might confuse developers because options & env vars are related. While verbose, maybe something super clear like `can_be_set_by_environment_target=True` or `environment_target=True`?

Reply:
I think that I'm fine with a short name, since how we explain what this means is very different based on audience:
1. plugin authors should look at a docstring somewhere
2. end users should get a more human-readable/linked explanation via the docsite

But maybe `environment_specific` rather than "sensitive".